src/vsr/replica.zig

```diff
@@ -3477,9 +3622,25 @@ pub fn ReplicaType(
             }
         }
 
-        fn on_request_read_reply_callback(client_replies: *ClientReplies, reply_: ?*Message) void {
+        fn on_request_read_reply_callback(
+            client_replies: *ClientReplies,
+            reply_header: *const Header,
+            reply_: ?*Message,
+            destination_replica: ?u8,
+        ) void {
             const self = @fieldParentPtr(Self, "client_replies", client_replies);
-            const reply = reply_ orelse return;
+            assert(destination_replica == null);
+
+            const reply = reply_ orelse {
+                if (self.client_sessions().get_slot_for_header(reply_header)) |slot| {
+                    self.client_replies.faulty.set(slot.index);
```

Comment:
Huh, it's a bit too subtle that we set faulty outise of `client_replies`. I am wondering, can we just unconditionally set faulty in `client_replies`? 

If we have updated the header for this client already, than we also should have a write queued, and that would clean the faulty bit?

ahhh.... that's not the case --- if the reply is header only, then we won't queue the write, and won't set the faulty bit.

Reply:
Yeah, ClientReplies doesn't know which header should be in the slot. If a lagging replica requests reply X, but we have a newer reply Y, then we don't want to set `faulty` just because `X` is missing. We can't just compare the op of the reply that we read, because we don't write header-only messages. I could give `ClientReplies` access to `ClientSessions`, but I kind of like the separation. :thinking: